### ISSUE TITLE
Allow thickness of the bar cursor to be configured?

### ISSUE THREAD

**Issue by varmanishant** [eclipse-theia/theia#7168]
Latest `xterm.js` supports configuring the thickness of bar cursor via https://github.com/xtermjs/xterm.js/pull/2590. Is there a task existing already to upgrade `xterm.js`? If not we could create a new one and add this feature after it is resolved?

**Comment by vince-fugnitto** [comment]
> Can we upgrade `xterm.js` to get this and other features?

@varmanishant is it something you'd like to take care of?

**Comment by varmanishant** [comment]
@vince-fugnitto , Thanks for the response. I am trying to go through it - will let you know on my progress. As, I am new, I need to figure out a few things.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto , Thanks for the response. I am trying to go through it - will let you know on my progress. As, I am new, I need to figure out a few things.

No problem :) Don't hesitate to ask if you have any questions!

**Comment by vince-fugnitto** [comment]
Related https://github.com/eclipse-theia/theia/pull/7121

**Comment by varmanishant** [comment]
Thanks for adding this PR @vince-fugnitto . I just added my thoughts on the PR. Basically it is the same as what I did; however I commented out a block which seemed to fail ... I am going to re-check from scratch to familiarize with the build etc.

**Comment by varmanishant** [comment]
Can someone let me know the exact way to develop a package say @theia/terminal?

I am doing `npx watch @theia/terminal` and modifying the code. When I see a change detected, I stop watching and do `yarn build` to reflect the changes. Is this correct or am I missing something? I always find a cache problem with the old files loading in the browser unless I I do run `yarn` which takes a lot of time. (Especially it tries to download vsix and fails probably due to Github Token or something like that.)

**Comment by vince-fugnitto** [comment]
> Can someone let me know the exact way to develop a package

You can use the following approach to make changes, and update the application.
Let's say I am updating `@theia/terminal` in the repository like you mentioned:

**Build Approach**
1. make changes to `@theia/terminal`
2. build my changes:
   ```shell
   npx run build @theia/terminal # build the terminal ext
   npx run build @theia/example-browser # build the example-browser
   yarn start:browser # re-start the backend
   ```

**Watch Approach**
1. given that the repo is built
2. watch the `@theia/terminal` ext
   ```shell
    npx run watch @theia/terminal # watches for changes in the terminal ext
   ```
3. watch the `@theia/example-browser` ext
    ```shell
    npx run watch @theia/example-browser # watches for changes from other ext (terminal)
    ```
4. start the application
    ```
    yarn start:browser
    ```
5. make changes to `@theia/terminal` (the changes should be watched)
6. re-load the frontend (browser) on new changes

**Comment by varmanishant** [comment]
Ah that explains it, thanks @vince-fugnitto. I was missing the `@theia/example-browser` part!

**Comment by varmanishant** [comment]
Refer: https://github.com/eclipse-theia/theia/pull/7284

**Comment by varmanishant** [comment]
There is one more feature that needs to be added: Terminal themes are not reflecting even though there is a code to do that. Should we create a new ticket for that?

I can see that in [terminal-theme-service.ts](https://github.com/eclipse-theia/theia/blob/master/packages/terminal/src/browser/terminal-theme-service.ts#L180-L182) we are taking the substring from the 13th position and that doesn't just work because the `colorname` comes as "Black, BrightBlack etc." which is not compatible with `xterm.js` that needs it as "black, brightBlack etc.". The other problem is to read it from the settings ... and we need to decide if it belongs in this file or the instance creation file.

**Comment by vince-fugnitto** [comment]
> There is one more feature that needs to be added:- terminal themes are not reflecting even though there is a code to do that. Should we create a new ticket for that?
> 
> In [terminal-theme-service.ts](https://github.com/eclipse-theia/theia/blob/924e6a2f201cb3609b48903118d55762369222e2/packages/terminal/src/browser/terminal-theme-service.ts#L180-L182) we are taking the substring from the 13th position; but that doesn't just work because the `colorname` comes as "Black, BrightBlack etc." which is not compatible with `xterm.js` which needs it as "black, brightBlack etc."? The other problem is that we need to read it from the user settings as well. But does that logic belongs tothis file or the instance creation file [terminal-widget-impl.ts](https://github.com/eclipse-theia/theia/blob/924e6a2f201cb3609b48903118d55762369222e2/packages/terminal/src/browser/terminal-widget-impl.ts)?

I believe the following issue tracks the theming not working correctly for terminals:
#7280

**Comment by vince-fugnitto** [comment]
Closed by #7284
